util/dex/config.go

```diff
@@ -35,8 +35,17 @@ func GenerateDexConfigYAML(settings *settings.ArgoCDSettings) ([]byte, error) {
 	dexCfg["telemetry"] = map[string]interface{}{
 		"http": "0.0.0.0:5558",
 	}
-	dexCfg["oauth2"] = map[string]interface{}{
-		"skipApprovalScreen": true,
+
+	oauth2Cfg, ok := dexCfg["oauth2"].(map[string]interface{})
+	if ok {
+		_ , ok := oauth2Cfg["skipApprovalScreen"].(bool)
+		if !ok {
+			oauth2Cfg["skipApprovalScreen"] = true
+		}
+	}else {
+		dexCfg["oauth2"] = map[string]interface{}{
+			"skipApprovalScreen": true,
+		}
 	}
 
```

Comment:
It looks good.  I would slightly simplify it.
`	if oauth2Cfg, found := dexCfg["oauth2"].(map[string]interface{}); found {
		if _, found := oauth2Cfg["skipApprovalScreen"].(bool); !found {
			oauth2Cfg["skipApprovalScreen"] = true
		}
	} else {
		dexCfg["oauth2"] = map[string]interface{}{
			"skipApprovalScreen": true,
		}
	}`

Reply:
Thanks for the advice.  I'm new to Go and not used to variable declaration inside of an if-statement.